src/Illuminate/Support/Collection.php

```diff
@@ -522,11 +522,16 @@ public function pop()
      * Push an item onto the beginning of the collection.
      *
      * @param  mixed  $value
+     * @param  mixed  $key
      * @return $this
      */
-    public function prepend($value)
+    public function prepend($value, $key = null)
     {
-        array_unshift($this->items, $value);
+        if (is_null($key)) {
+            array_unshift($this->items, $value);
+        } else {
+            $this->items = [$key => $value] + $this->items;
```

Comment:
array_merge is not the same as the plus, which is why array_merge is used elsewhere.

In this case it would result in the same, but I like way the plus looks; it's much cleaner-looking than array_merge.